--- Makefile_exe	2025-10-04 12:44:37.571741347 -0400
+++ Makefile_exe_fixed	2025-10-06 11:11:08.201056897 -0400
@@ -5,45 +5,49 @@
 
 WORKDIR = $PWD
 
-CC = gcc
-CXX = g++
-AR = ar
-LD = g++
-WINDRES = windres
+COMPILER_PREFIX = i686-w64-mingw32-
+
+CC = $(COMPILER_PREFIX)gcc
+CXX = $(COMPILER_PREFIX)g++
+AR = $(COMPILER_PREFIX)ar
+LD = $(COMPILER_PREFIX)g++
+WINDRES = $(COMPILER_PREFIX)windres
 
 INC = 
 CFLAGS = -std=c++11 -pedantic -Wall -Wextra -Wshadow -Wundef -Wuseless-cast -Wpointer-arith -Wfloat-conversion
 RCFLAGS = 
 RESINC = 
 LIBDIR = 
-LIB = 
-LDFLAGS = 
+LIB = -luser32
+LDFLAGS = -static-libgcc -static-libstdc++ -static
 
 INC_DEBUG = $(INC)
 CFLAGS_DEBUG = $(CFLAGS) -g -Og -DBUILD_DEBUG
 RESINC_DEBUG = $(RESINC)
 RCFLAGS_DEBUG = $(RCFLAGS)
 LIBDIR_DEBUG = $(LIBDIR)
-LIB_DEBUG = $(LIB)user32
+LIB_DEBUG = $(LIB)
 LDFLAGS_DEBUG = $(LDFLAGS)
 OBJDIR_DEBUG = build/obj/debug
 DEP_DEBUG = 
-OUT_DEBUG = build/bin/debug/thinker
+OUT_DEBUG = build/bin/debug/thinker.exe
 
 INC_RELEASE = $(INC)
 CFLAGS_RELEASE = $(CFLAGS) -O2 -flto -fno-rtti -fno-strict-aliasing -fno-delete-null-pointer-checks -DBUILD_REL
 RESINC_RELEASE = $(RESINC)
 RCFLAGS_RELEASE = $(RCFLAGS)
 LIBDIR_RELEASE = $(LIBDIR)
-LIB_RELEASE = $(LIB)user32
+LIB_RELEASE = $(LIB)
 LDFLAGS_RELEASE = $(LDFLAGS) -s -static
 OBJDIR_RELEASE = build/obj/release
 DEP_RELEASE = 
-OUT_RELEASE = build/bin/release/thinker
+OUT_RELEASE = build/bin/release/thinker.exe
 
-OBJ_DEBUG = $(OBJDIR_DEBUG)/src/launch.o $(OBJDIR_DEBUG)/src/launch.o
+OBJ_DEBUG = $(OBJDIR_DEBUG)/src/launch.o
+OBJ_RELEASE = $(OBJDIR_RELEASE)/src/launch.o
 
-OBJ_RELEASE = $(OBJDIR_RELEASE)/src/launch.o $(OBJDIR_RELEASE)/src/launch.o
+RES_DEBUG = $(OBJDIR_DEBUG)/src/launch.res
+RES_RELEASE = $(OBJDIR_RELEASE)/src/launch.res
 
 all: before_build build_debug build_release after_build
 
@@ -52,7 +56,6 @@
 before_build: 
 
 after_build: 
-	cmd /c copy "$(PROJECT_DIR)$(TARGET_OUTPUT_FILE)" patch\
 
 before_debug: 
 	test -d build/bin/debug || mkdir -p build/bin/debug
@@ -64,12 +67,14 @@
 
 debug: before_build build_debug after_build
 
-out_debug: before_debug $(OBJ_DEBUG) $(DEP_DEBUG)
-	$(LD) $(LIBDIR_DEBUG) -o $(OUT_DEBUG) $(OBJ_DEBUG)  $(LDFLAGS_DEBUG) $(LIB_DEBUG)
+out_debug: before_debug $(OBJ_DEBUG) $(RES_DEBUG) $(DEP_DEBUG)
+	$(LD) $(LIBDIR_DEBUG) -o $(OUT_DEBUG) $(OBJ_DEBUG) $(RES_DEBUG)  $(LDFLAGS_DEBUG) $(LIB_DEBUG)
 
 $(OBJDIR_DEBUG)/src/launch.o: src/launch.cpp src/launch.rc
 	$(CXX) $(CFLAGS_DEBUG) $(INC_DEBUG) -c src/launch.cpp -o $(OBJDIR_DEBUG)/src/launch.o
-	$(CPP) $(RCFLAGS_DEBUG) -i src/launch.rc -o $(OBJDIR_DEBUG)/src/launch.o -O coff $(RESINC_DEBUG)
+
+$(OBJDIR_DEBUG)/src/launch.res: src/launch.rc
+	$(WINDRES) src/launch.rc -O coff -o $(OBJDIR_DEBUG)/src/launch.res
 
 clean_debug: 
 	rm -f $(OBJ_DEBUG) $(OUT_DEBUG)
@@ -86,12 +91,14 @@
 
 release: before_build build_release after_build
 
-out_release: before_release $(OBJ_RELEASE) $(DEP_RELEASE)
-	$(LD) $(LIBDIR_RELEASE) -o $(OUT_RELEASE) $(OBJ_RELEASE)  $(LDFLAGS_RELEASE) $(LIB_RELEASE)
+out_release: before_release $(OBJ_RELEASE) $(RES_RELEASE) $(DEP_RELEASE)
+	$(LD) $(LIBDIR_RELEASE) -o $(OUT_RELEASE) $(OBJ_RELEASE) $(RES_RELEASE)  $(LDFLAGS_RELEASE) $(LIB_RELEASE)
 
 $(OBJDIR_RELEASE)/src/launch.o: src/launch.cpp src/launch.rc
 	$(CXX) $(CFLAGS_RELEASE) $(INC_RELEASE) -c src/launch.cpp -o $(OBJDIR_RELEASE)/src/launch.o
-	$(CPP) $(RCFLAGS_RELEASE) -i src/launch.rc -o $(OBJDIR_RELEASE)/src/launch.o -O coff $(RESINC_RELEASE)
+
+$(OBJDIR_RELEASE)/src/launch.res: src/launch.rc
+	$(WINDRES) src/launch.rc -O coff -o $(OBJDIR_RELEASE)/src/launch.res
 
 clean_release: 
 	rm -f $(OBJ_RELEASE) $(OUT_RELEASE)
