--- Makefile_dll.orig	2021-12-13 23:25:39.012697527 +0100
+++ Makefile_dll	2021-12-13 02:01:27.737305779 +0100
@@ -5,10 +5,12 @@
 
 WORKDIR = `pwd`
 
-CC = gcc
-CXX = g++
-AR = ar
-LD = g++
+COMPILER_PREFIX = i686-w64-mingw32-
+
+CC = $(COMPILER_PREFIX)gcc
+CXX = $(COMPILER_PREFIX)g++
+AR = $(COMPILER_PREFIX)ar
+LD = $(COMPILER_PREFIX)g++
 WINDRES = 
 
 INC = 
@@ -16,7 +18,7 @@
 RESINC = 
 LIBDIR = 
 LIB = -luser32 -lwinmm -lgdi32 -lpsapi
-LDFLAGS = 
+LDFLAGS = -static-libgcc -static-libstdc++ -static
 
 INC_DEBUG = $(INC)
 CFLAGS_DEBUG = $(CFLAGS) -g -Og -DBUILD_DLL -DBUILD_DEBUG
@@ -29,7 +31,7 @@
 DEP_DEBUG = 
 DEF_DEBUG = build/bin/debug/thinker.def
 SHAREDINT_DEBUG = build/bin/debug/libthinker.a
-OUT_DEBUG = build/bin/debug/thinker.so
+OUT_DEBUG = build/bin/debug/thinker.dll
 
 INC_DEVELOP = $(INC)
 CFLAGS_DEVELOP = $(CFLAGS) -O2 -fsplit-paths -fno-strict-aliasing -fno-delete-null-pointer-checks -DBUILD_DLL
@@ -42,7 +44,7 @@
 DEP_DEVELOP = 
 DEF_DEVELOP = build/bin/develop/thinker.def
 SHAREDINT_DEVELOP = build/bin/develop/libthinker.a
-OUT_DEVELOP = build/bin/develop/thinker.so
+OUT_DEVELOP = build/bin/develop/thinker.dll
 
 INC_RELEASE = $(INC)
 CFLAGS_RELEASE = $(CFLAGS) -O2 -fsplit-paths -fno-strict-aliasing -fno-delete-null-pointer-checks -DBUILD_DLL -DBUILD_REL
@@ -55,7 +57,7 @@
 DEP_RELEASE = 
 DEF_RELEASE = build/bin/release/thinker.def
 SHAREDINT_RELEASE = build/bin/release/libthinker.a
-OUT_RELEASE = build/bin/release/thinker.so
+OUT_RELEASE = build/bin/release/thinker.dll
 
 OBJ_DEBUG = $(OBJDIR_DEBUG)/src/move.o $(OBJDIR_DEBUG)/src/test.o $(OBJDIR_DEBUG)/src/terranx.o $(OBJDIR_DEBUG)/src/tech.o $(OBJDIR_DEBUG)/src/plan.o $(OBJDIR_DEBUG)/src/path.o $(OBJDIR_DEBUG)/src/patch.o $(OBJDIR_DEBUG)/src/base.o $(OBJDIR_DEBUG)/src/map.o $(OBJDIR_DEBUG)/src/main.o $(OBJDIR_DEBUG)/src/lib/ini.o $(OBJDIR_DEBUG)/src/gui.o $(OBJDIR_DEBUG)/src/game.o $(OBJDIR_DEBUG)/src/engine.o
 
@@ -70,7 +72,6 @@
 before_build: 
 
 after_build: 
-	cmd /c copy "$(PROJECT_DIR)$(TARGET_OUTPUT_FILE)" patch\
 
 before_debug: 
 	test -d build/bin/debug || mkdir -p build/bin/debug
@@ -117,7 +118,7 @@
 	$(CXX) $(CFLAGS_DEBUG) $(INC_DEBUG) -c src/main.cpp -o $(OBJDIR_DEBUG)/src/main.o
 
 $(OBJDIR_DEBUG)/src/lib/ini.o: src/lib/ini.c
-	$(CPP) $(CFLAGS_DEBUG) $(INC_DEBUG) -c src/lib/ini.c -o $(OBJDIR_DEBUG)/src/lib/ini.o
+	$(CC) $(CFLAGS_DEBUG) $(INC_DEBUG) -c src/lib/ini.c -o $(OBJDIR_DEBUG)/src/lib/ini.o
 
 $(OBJDIR_DEBUG)/src/gui.o: src/gui.cpp
 	$(CXX) $(CFLAGS_DEBUG) $(INC_DEBUG) -c src/gui.cpp -o $(OBJDIR_DEBUG)/src/gui.o
@@ -179,7 +180,7 @@
 	$(CXX) $(CFLAGS_DEVELOP) $(INC_DEVELOP) -c src/main.cpp -o $(OBJDIR_DEVELOP)/src/main.o
 
 $(OBJDIR_DEVELOP)/src/lib/ini.o: src/lib/ini.c
-	$(CPP) $(CFLAGS_DEVELOP) $(INC_DEVELOP) -c src/lib/ini.c -o $(OBJDIR_DEVELOP)/src/lib/ini.o
+	$(CC) $(CFLAGS_DEVELOP) $(INC_DEVELOP) -c src/lib/ini.c -o $(OBJDIR_DEVELOP)/src/lib/ini.o
 
 $(OBJDIR_DEVELOP)/src/gui.o: src/gui.cpp
 	$(CXX) $(CFLAGS_DEVELOP) $(INC_DEVELOP) -c src/gui.cpp -o $(OBJDIR_DEVELOP)/src/gui.o
@@ -241,7 +242,7 @@
 	$(CXX) $(CFLAGS_RELEASE) $(INC_RELEASE) -c src/main.cpp -o $(OBJDIR_RELEASE)/src/main.o
 
 $(OBJDIR_RELEASE)/src/lib/ini.o: src/lib/ini.c
-	$(CPP) $(CFLAGS_RELEASE) $(INC_RELEASE) -c src/lib/ini.c -o $(OBJDIR_RELEASE)/src/lib/ini.o
+	$(CC) $(CFLAGS_RELEASE) $(INC_RELEASE) -c src/lib/ini.c -o $(OBJDIR_RELEASE)/src/lib/ini.o
 
 $(OBJDIR_RELEASE)/src/gui.o: src/gui.cpp
 	$(CXX) $(CFLAGS_RELEASE) $(INC_RELEASE) -c src/gui.cpp -o $(OBJDIR_RELEASE)/src/gui.o
